File: src/Entity/Contrat.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * Contrat
 *
 * @ORM\Table(name="contrat", indexes={@ORM\Index(name="IdEST", columns={"idEST"})})
 * @ORM\Entity
 */
class Contrat
{
    /**
     * @var int
     *
     * @ORM\Column(name="idC", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idc;

    /**
     * @ORM\Column(name="nomC", type="string", length=255, nullable=false)
     * @Assert\Length(min=5)
     * @Assert\Regex(
     *     pattern="/^[a-zA-Z]+$/",
     *     message="Le nom ne doit contenir que des lettres"
     * )
     */
    private $nomc;

    
    /**
     * @ORM\Column(name="DateDC", type="string", length=255, nullable=false)
     * @Assert\Date
     */
    private $datedc;

     /**
     * @ORM\Column(name="DateFC", type="string", length=255, nullable=false)
     * @Assert\Date
     * @Assert\Expression(
     *     "this.getDatedc() < this.getDatefc()",
     *     message="La date de fin doit être après la date de début"
     * )
     */
    private $datefc;

    /**
     * @var \Espacetalent
     *
     * @ORM\ManyToOne(targetEntity="Espacetalent")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idEST", referencedColumnName="idEST")
     * })
     */
    private $idest;

    public function getIdc(): ?int
    {
        return $this->idc;
    }

    public function getNomc(): ?string
    {
        return $this->nomc;
    }

    public function setNomc(string $nomc): self
    {
        $this->nomc = $nomc;

        return $this;
    }

    public function getDatedc(): ?string
    {
        return $this->datedc;
    }

    public function setDatedc(string $datedc): self
    {
        $this->datedc = $datedc;

        return $this;
    }

    public function getDatefc(): ?string
    {
        return $this->datefc;
    }

    public function setDatefc(string $datefc): self
    {
        $this->datefc = $datefc;

        return $this;
    }

    public function getIdest(): ?Espacetalent
    {
        return $this->idest;
    }

    public function setIdest(?Espacetalent $idest): self
    {
        $this->idest = $idest;

        return $this;
    }
    public function __toString(): string
    {
        return $this->nomc;
    }

}

File: AdoDocMobile/src/com/mycompany/myapp/entities/Contrat.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author ASMA
 */
public class Contrat {
    private int idC;
    private String nomC,DateDC,DateFC;
 private int idEST;
    public Contrat(String nomC) {
        this.nomC = nomC;
    }

    public Contrat() {
    }

    public Contrat(int idC, String nomC, String DateDC, String DateFC, int idEST) {
        this.idC = idC;
        this.nomC = nomC;
        this.DateDC = DateDC;
        this.DateFC = DateFC;
        this.idEST = idEST;
    }

    public Contrat(String nomC, String DateDC, String DateFC, int idEST) {
        this.nomC = nomC;
        this.DateDC = DateDC;
        this.DateFC = DateFC;
        this.idEST = idEST;
    }

    public String getDateDC() {
        return DateDC;
    }

    public int getIdC() {
        return idC;
    }

    public String getDateFC() {
        return DateFC;
    }

    public int getIdEST() {
        return idEST;
    }

    public String getNomC() {
        return nomC;
    }

    public void setDateDC(String DateDC) {
        this.DateDC = DateDC;
    }

    public void setDateFC(String DateFC) {
        this.DateFC = DateFC;
    }

    public void setIdC(int idC) {
        this.idC = idC;
    }

    public void setIdEST(int idEST) {
        this.idEST = idEST;
    }

    public void setNomC(String nomC) {
        this.nomC = nomC;
    }

    @Override
    public String toString() {
        return "Contrat{" + "idC=" + idC + ", nomC=" + nomC + ", DateDC=" + DateDC + ", DateFC=" + DateFC + ", idEST=" + idEST + '}';
    }

    
    
}
